pandas/core/groupby.py

```diff
@@ -3719,7 +3772,12 @@ def _wrap_agged_blocks(self, items, blocks):
         if self.axis == 1:
             result = result.T
 
-        return self._reindex_output(result)._convert(datetime=True)
+        if isinstance(self.inputconstructor(), SparseDataFrame):
+            return DataFrame(
```

Comment:
again this is a completely crazy thing. doing this kind of introspection is so fraught with peril and should simply not be done here. Yes maybe 1 function to do it is fine.